src/main/java/gregtech/api/recipes/CountableIngredient.java

```diff
@@ -11,9 +12,17 @@
 import net.minecraftforge.oredict.OreIngredient;
 
 import java.util.Arrays;
-import java.util.Objects;
 
-public class CountableIngredient {
+public class CountableIngredient<C extends CountableIngredient<C>> {
+
+    private final Ingredient ingredient;
+    private final int count;
+    private boolean nonConsumable = false;
+
+    public CountableIngredient(Ingredient ingredient, int count) {
+        this.ingredient = ingredient;
+        this.count = count;
```

Comment:
This causes major issues with some of our autogen recipes still setting up ingredient count at 0 (distillery, fluid heater, bending machine... might need to revisit the `IntCircuitRecipeBuilder#finalizeAndValidate` method).
Having an ingredient of count 0 with this new system causes the ItemStack to disappear from JEI previews, and causes parallel to divide by 0 (crash).
```suggestion
        if (count <= 0) {
            this.count = 1;
            setNonConsumable();
        } else {
            this.count = count;
        }
```
This fixes it and ensures it will not break if anything else sets `count` to 0. If you have a more clever solution like throwing a warning go ahead.

Reply:
This is just `IntCircuitRecipeIngredient` needing its `finalizeAndValidate` call updated. Right now it is still passing in a count of `0`

Reply:
Yes, however keeping this constructor as is will not throw any error, and yet will end up causing issues for whatever else sets an ingredient with a count of 0, be it an addon or CT script.